baremetal_network_provisioning/ml2/extensions/switch.py

```diff
@@ -12,29 +12,225 @@
 #    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 #    License for the specific language governing permissions and limitations
 #    under the License.from oslo.config import cfg
+
+from simplejson import scanner as json_scanner
+
 from neutron.api import extensions
+from neutron.api.v2 import attributes
+from neutron import context
 from neutron import wsgi
 
+from baremetal_network_provisioning.common import constants as const
+from baremetal_network_provisioning.common import exceptions as exc
+from baremetal_network_provisioning.db import bm_nw_provision_db as db
+from baremetal_network_provisioning.drivers import discovery_driver
+
+from oslo_log import log as logging
+
+LOG = logging.getLogger(__name__)
+
 
-EXTENDED_ATTRIBUTES_2_0 = {
-    "switches": {
-    }
+RESOURCE_ATTRIBUTE_MAP = {
+    'switches': {
+        'id': {'allow_post': False, 'allow_put': False,
+               'is_visible': True},
+        'ip_address': {'allow_post': True, 'allow_put': False,
```

Comment:
are we allowing the hostname for the switch also ? if not change the validate (string to 'validate': {'type:ip_address_or_none') type. simple string validation might not be sufficient